ui/src/shared/components/Dygraph.js

```diff
@@ -106,55 +103,66 @@ export default React.createClass({
         strokeWidth: 2,
         highlightCircleSize: 5,
       },
-      highlightCallback(e, x, points) {
+      unhighlightCallback: () => {
+        legendContainerNode.className = 'container--dygraph-legend hidden' // hide
+
+        // this.isMouseOverGraph = false
+      },
+      highlightCallback: e => {
+        // must show before drawing so that dimension calcs below work
+        legendContainerNode.className = 'container--dygraph-legend' // show
+
         // Move the Legend on hover
         const graphRect = graphContainerNode.getBoundingClientRect()
         const legendRect = legendContainerNode.getBoundingClientRect()
         const graphWidth = graphRect.width + 32 // Factoring in padding from parent
+        const graphHeight = graphRect.height
+        const graphBottom = graphRect.bottom
         const legendWidth = legendRect.width
+        const legendHeight = legendRect.height
+        const screenHeight = window.innerHeight
         const legendMaxLeft = graphWidth - legendWidth / 2
         const trueGraphX = e.pageX - graphRect.left
-        const legendTop = graphRect.height + 0
+
         let legendLeft = trueGraphX
+
         // Enforcing max & min legend offsets
         if (trueGraphX < legendWidth / 2) {
           legendLeft = legendWidth / 2
         } else if (trueGraphX > legendMaxLeft) {
           legendLeft = legendMaxLeft
         }
 
+        // Disallow screen overflow of legend
+        const legendBottomClipped = graphBottom + legendHeight > screenHeight
```

Comment:
Yep, works as intended -- addition is happening prior to comparison. Prettier actually removed parens if I try to add them 😈